### ISSUE TITLE
Ads being injected after a page loads

### ISSUE THREAD

**Issue by Kraust** [gorhill/uBlock#202]
Example site: http://www.speedtest.net/

This seems to be an issue with uBlock and has been there for a while. I know that you're really adamant on saying that you block everything ABP does, but this seems to be an edge case where that isn't true (I know that ABP / AdBlock block these kinds of ads)

I have standard filter lists set: http://puu.sh/hOFfx.png
However when I visit some sites with these dynamic ads I get something like this: http://puu.sh/hOFfx.png

I am 99% sure I am doing nothing wrong. I've been using browser based ad blockers for almost a decade now.

**Comment by gorhill** [comment]
I can't reproduce, I don't see any ads, using your filter lists. There must be something more on your side which I don't have, as I can't reproduce. I see 5 filters in your custom filter list. What are they?

The link to your second picture is the same as the link of the first picture -- the filter lists.

**Comment by iandol** [comment]
I don't see ads on speedtest.net either, uBo V0.9.7...

**Comment by gorhill** [comment]
There was an [update of EasyList some days ago regarding `speedtest.net`](https://hg.adblockplus.org/easylist/rev/411cba03e74a). Are the filters that were added in your copy of EasyList?

**Comment by Kraust** [comment]
I purged my cache and re-got the Filter List and this issue is still present.

The custom filters weren't anything much (I had one to manually filter this but I've since removed it)

    www.gamefaqs.com###content.container > .main_content.row > .span8 > .pod.body.board_wrap >       .body > .board.topics.othertopics
    www.gamefaqs.com###footer.footer
    *s.yimg.com/uq/syndication/*

And sorry the picture of the ads
![ss 2015-05-15 at 07 10 52](https://cloud.githubusercontent.com/assets/6548135/7663297/2f95781e-fb36-11e4-84f8-88844bab7322.png)

**Comment by gorhill** [comment]
Can you select that banner with the element picker? I would like to see what are the choices of filters offered.

**Comment by gorhill** [comment]
What's your browser/version, uBlock version?

**Comment by Kraust** [comment]
![ss 2015-05-15 at 07 15 07](https://cloud.githubusercontent.com/assets/6548135/7663328/be2b4ebe-fb36-11e4-89a8-cf1e2ce18e98.png)

I had made a custom filter for it just to remove the div contained in it:
    ##.dfp-ab-leaderboard

It's very weird because selecting the element with the picker doesn't select it. Instead it selects some Rogue Javascript:

![ss 2015-05-15 at 07 17 43](https://cloud.githubusercontent.com/assets/6548135/7663354/1a77d2c8-fb37-11e4-8dde-24fd3660eccd.png)

**Comment by Kraust** [comment]
I want to point out that the issue is most likely the rogue iframe. I've had issues with them containing ads before.

**Comment by gorhill** [comment]
In the element picker choices, I can get up to `#dfp-ab-leaderboard` as seen in your screenshot, after this, there is nothing -- I have an empty `div`. So something added DOM elements in there.

I would look in the log to see where that banner's come from. On my side, most of what is blocked are scripts.

Note that your filter lists is not something you can use on ABP, so this is not a strict comparison. Pick only EasyList (as would happen with ABP) and see the result.

**Comment by gorhill** [comment]
This is what got blocked on my side:

    19:19:52    --  ||pipedream.wistia.com^  xhr  http://pipedream.wistia.com/mput?topic=metrics
    19:19:51    --  ||distillery.wistia.com^  script  http://distillery.wistia.com/x?data=ewoiYWNjb3VudF9rZXkiOiJ3aXN0aWEtcHJvZHVjdGlvbl8xMzExMzUiLAoic2Vzc2lvbl9pZCI6InYyMDE1MDIyN185MGIzOWU0OC0yZGIzLTQxZWQtYTAxNi00NDZkMzY2MGEyZTciLAoibWVkaWFfaWQiOiJ3aXN0aWEtcHJvZHVjdGlvbl84NjU2Mzg5IiwKImV2ZW50X2tleSI6InYyMDE1MDIyNV83OTk3MWU0OC1jNzkyLTQ1ZjMtOTMxYi0yNzI1ZjNjN2E2MmMiLAoibWVkaWFfZHVyYXRpb24iOjEzOS40NCwKInZpc2l0b3JfdmVyc2lvbiI6MCwKInJlZmVycmVyIjoiaHR0cDovL3d3dy5zcGVlZHRlc3QubmV0LyIsImV2ZW50X2RldGFpbHMiOlt7ImtleSI6ImluaXRpYWxpemVkIiwidmFsdWUiOiItMS4wIiwidGltZURlbHRhIjo0MiwibGFzdEFjY291bnRJbnN0YW5jZSI6Im5vbmUiLCJsYXN0TWVkaWFJbnN0YW5jZSI6Im5vbmUifV19
    19:19:51    --  ##.lbc-ad  dom  http://www.speedtest.net/
    19:19:51    --  ##.lbc  dom  http://www.speedtest.net/
    19:19:51    --  ##.alt-promo-container > ul > .alt-promo:first-child + .alt-promo  dom  http://www.speedtest.net/
    19:19:51    --  ##.ad-vertical-stack-ad  dom  http://www.speedtest.net/
    19:19:51    --  ##.ad-vertical  dom  http://www.speedtest.net/
    19:19:51    --  ##.ad-space  dom  http://www.speedtest.net/
    19:19:51    --  ###footer-social  dom  http://www.speedtest.net/
    19:19:51    --  ###ad-top-wrap  dom  http://www.speedtest.net/
    19:19:51    --  ||secure-us.imrworldwide.com^  script  http://secure-us.imrworldwide.com/v60.js
    19:19:51    --  ||b.scorecardresearch.com^  script  http://b.scorecardresearch.com/beacon.js
    19:19:51    --  ||c.betrad.com^  script  http://c.betrad.com/geo/c.js
    19:19:51    --  ||googletagmanager.com/gtm.js?  script  http://www.googletagmanager.com/gtm.js?id=GTM-MX6CBT
    19:19:51    --  ||www.googletagservices.com^  script  http://www.googletagservices.com/tag/js/gpt.js
    19:19:51    --  ||ziffdavis.com^  script  http://static.ziffdavis.com/js/zddmp/1.0/zddmp.min.js
    19:19:51    --  ||ads.ookla.com^  script

**Comment by Kraust** [comment]
Hmm, I disabled uBlockO and enabled AdBlockPlus. This isn't Blocked by AdBlockPlus (With just EasyList) either. That's throws all sorts of Red Flags on me doing something wrong.

How do I view the list of blocked elements? I tried looking at the Network Requests, but I don't know how to filter through that.

**Comment by gorhill** [comment]
Maybe you just stumbled on a new ad which will require a new filter (just like new filters were required on May 6 in EasyList). Since you can reproduce with ABP, best is probably you report to ABP.

Open the request logger and then force a refresh of the page. All requests will be listed, Now is to find the one corresponding to the ad, it's pulled from somewhere and the key is to find from where.

**Comment by Kraust** [comment]
Well I just noticed something strange.
This ad only appears when *refreshing* the page. On a normal page load it will not appear.

Also the domain is rateaccept.net. I am not sure exactly which request is the one that does it as there seem to be a dozen different embedded divs / iframes Perhaps try to replicate this?

1. Go to http://www.speedtest.net
2. Wait for the page to load
3. Reload the page (F5 / Reload Button)

I just added rateaccept.net to my filter list to get rid of the ad.
